core/src/main/scala/org/apache/spark/util/collection/ExternalAppendOnlyMap.scala

```diff
@@ -463,21 +463,21 @@ class ExternalAppendOnlyMap[K, V, C](
 
     // An intermediate stream that reads from exactly one batch
     // This guards against pre-fetching and other arbitrary behavior of higher level streams
-    private var deserializeStream = nextBatchStream()
+    private var deserializeStream = null.asInstanceOf[Option[DeserializationStream]]
```

Comment:
Much better: `... deserializeStream: Option[DeserializationStream] = None`. It's not just the cast, but the fact that you're assigning null to an Option